File: CampusLinkBackend/src/main/java/com/campuslink/campuslinkbackend/constants/RegistrationStatus.java
package com.campuslink.campuslinkbackend.constants;

public final class RegistrationStatus {

    private RegistrationStatus() {}

    public static final String ACTIVE = "ACTIVE";
    public static final String PENDING_PAYMENT = "PENDING_PAYMENT";
    public static final String PENDING_PAYMENT_VERIFICATION = "PENDING_PAYMENT_VERIFICATION";
    public static final String PAYMENT_APPROVED = "PAYMENT_APPROVED";
    public static final String PAYMENT_REJECTED = "PAYMENT_REJECTED";
    public static final String PENDING_TEAM = "PENDING_TEAM";
    public static final String PENDING_INVITE = "PENDING_INVITE";
    public static final String REJECTED = "REJECTED";
    public static final String CANCELLED = "CANCELLED";
}

File: CampusLinkFrontend/campuslink-frontend/src/utils/registrationStatus.js
export const REGISTRATION_STATUS_LABELS = {
  ACTIVE: 'Registered',
  PENDING_PAYMENT_VERIFICATION: 'Pending Approval',
  PENDING_TEAM: 'Pending Team',
  PAYMENT_REJECTED: 'Payment Rejected',
  PENDING_INVITE: 'Invitation Pending',
  PENDING_PAYMENT: 'Pending Approval',
};

export const PENDING_REGISTRATION_STATUSES = [
  'PENDING_PAYMENT_VERIFICATION',
  'PENDING_TEAM',
  'PENDING_PAYMENT',
  'PENDING_INVITE',
];

export function registrationStatusLabel(status) {
  return REGISTRATION_STATUS_LABELS[status] || status;
}
